lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -859,4 +868,79 @@ public static function resolve_theme_file_uris( $theme_json ) {
 
 		return $theme_json;
 	}
+
+	/**
+	 * Adds variations sourced from block style variations files to the supplied theme.json data.
+	 *
+	 * @since 6.6.0
+	 *
+	 * @param array $data   Array following the theme.json specification.
+	 * @param array $variations Shared block style variations.
+	 * @return array Theme json data including shared block style variation definitions.
+	 */
+	private static function inject_variations_from_block_style_variation_files( $data, $variations ) {
```

Comment:
There's a subtle bug in this method:

- The block `core/heading` has a variation called 'Mine'.
- A partial defines a variation 'Mine' for the blocks `core/paragraph` and `core/group`.
- With this code, the styles the partial defined for `core/paragraph` and `core/group` will also apply to `core/heading`.

We'd need to 1) register the variations defined in the partial at the block level (`styles.blocks.blockType.variations`) but 2) make sure the top-level variation of the `theme.json` still override them. See `inject_variations_from_block_styles_registry` as an example.

Reply:
So the bug was caused by the shift to omit the `blockTypes` prop which was after this function had been written?

Reply:
No, the partial still has the `blockTypes` but this method doesn't use it. It just stores the variations found in partials in `styles.variations` instead of using the `blockTypes` to assign them to the specific blocks — that's the bug.

Reply:
I also should have been clearer in my question earlier. I meant it more that, previously the shared definitions still had the `blockTypes` prop which was used in the unwrapping of those definitions into the individual block types' variations. When the unwrapping was updated to unwrap shared definitions based on what was registered in the registry instead of `blockTypes`, this function was missed.

Reply:
There was also an issue that block styles contained in a block type's `styles` property, e.g. Button's Outline, weren't in the registry so shared definitions for them weren't unwrapped correctly.

I believe this has all been fix in recent days, so I'll resolve this thread pending further review.